src/main/java/frc/robot/commands/SwerveJoystickCommand.java

```diff
@@ -0,0 +1,123 @@
+
+import com.ctre.phoenix6.mechanisms.swerve.SwerveModule.DriveRequestType;
+import com.ctre.phoenix6.mechanisms.swerve.SwerveRequest;
+import edu.wpi.first.math.filter.SlewRateLimiter;
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.Constants;
+import frc.robot.subsystems.SwerveSubsystem;
+import frc.robot.util.SwerveRequestFieldCentric;
+import java.util.function.Supplier;
+
+public class SwerveJoystickCommand extends Command {
+  private final Supplier<Double> xSpdFunction,
+      ySpdFunction,
+      turningSpdFunction,
+      speedIncreaseControlFunction,
+      speedDecreaseControlFunction;
+
+  // Limits rate of change (in this case x, y, and turning movement)
+  private final SlewRateLimiter xLimiter, yLimiter, turningLimiter;
+
+  private final SwerveSubsystem swerveDrivetrain;
+
+  // Sets everything
+  public SwerveJoystickCommand(
+      Supplier<Double> frontBackFunction,
+      Supplier<Double> leftRightFunction,
+      Supplier<Double> turningSpdFunction,
+      Supplier<Double> speedIncreaseControlFunction,
+      Supplier<Double> speedDecreaseControlFunction,
+      SwerveSubsystem csd) {
+
+    this.xSpdFunction = frontBackFunction;
+    this.ySpdFunction = leftRightFunction;
+    this.turningSpdFunction = turningSpdFunction;
+    this.speedIncreaseControlFunction = speedIncreaseControlFunction;
+    this.speedDecreaseControlFunction = speedDecreaseControlFunction;
+    this.xLimiter = new SlewRateLimiter(Constants.Swerve.kTeleDriveMaxAccelerationUnitsPerSecond);
+    this.yLimiter = new SlewRateLimiter(Constants.Swerve.kTeleDriveMaxAccelerationUnitsPerSecond);
+    this.turningLimiter =
+        new SlewRateLimiter(Constants.Swerve.kTeleDriveMaxAngularAccelerationUnitsPerSecond);
+    this.swerveDrivetrain = csd;
+
+    // Adds the subsystem as a requirement (prevents two commands from acting on subsystem at once)
+    addRequirements(swerveDrivetrain);
+  }
+
+  @Override
+  public void initialize() {}
+
+  @Override
+  public void execute() {
+    // 1. Get real-time joystick inputs
+    double xSpeed = xSpdFunction.get(); // xSpeed is actually front back (front +, back -)
+    double ySpeed = ySpdFunction.get(); // ySpeed is actually left right (left +, right -)
+    double turningSpeed =
+        turningSpdFunction.get(); // turning speed is (anti-clockwise +, clockwise -)
+
+    // 2. Normalize inputs
+    double length = xSpeed * xSpeed + ySpeed * ySpeed; // acutally length squared
+    if (length > 1d) {
+      length = Math.sqrt(length);
+      xSpeed /= length;
+      ySpeed /= length;
+    }
+
+    // 3. Apply deadband
+    xSpeed = Math.abs(xSpeed) > Constants.OI.kLeftJoystickDeadband ? xSpeed : 0.0;
+    ySpeed = Math.abs(ySpeed) > Constants.OI.kLeftJoystickDeadband ? ySpeed : 0.0;
+    turningSpeed =
+        Math.abs(turningSpeed) > Constants.OI.kRightJoystickDeadband ? turningSpeed : 0.0;
+
+    // 4. Make the driving smoother
+    // Speed control
+    double driveSpeed =
```

Comment:
maybe add some sort of comment to describe whats going on? Or put into a private util function so that it reads like English and no comment is needed to understand whats going on

Reply:
English was used.